pkg/machines/components/createDialog.jsx

```diff
@@ -0,0 +1,347 @@
+        switch (key) {
+            case 'vendor': {
+                this.setState({
+                    [key]: value,
+                    "os": this.props.vendorMap[value][0].shortId
+                });
+                break;
+            }
+            case 'os':
+                this.setState({ [key]: value });
+                break;
+            case 'memorySize':
+                this.setState({ [key]: value });
+                value = toMegaBytes(value, this.state.memorySizeUnit)
+                break
+            case 'storageSize':
+                this.setState({ [key]: value });
+                value = toGigaBytes(value, this.state.storageSizeUnit)
+                break
+            case 'memorySizeUnit':
+                this.setState({ [key]: value });
+                value = toMegaBytes(this.state.memorySize, value)
+                key = 'memorySize'
+                break
+            case 'storageSizeUnit':
+                this.setState({ [key]: value });
+                value = toGigaBytes(this.state.memorySize, value)
+                key = 'storageSize'
+                break
+            default:
+                break;
+        }
+
+        if (this.props.valuesChanged) {
+            this.props.valuesChanged(key, value);
+        }
+    }
+
+
+    render () {
+        let vendors = (
+            this.props.vendorNames
+                .map(vendor => {
+                    return (<Select.SelectEntry data={vendor} key={vendor}>{vendor}</Select.SelectEntry>);
+                })
+        )
+
+        let osNames = (
+            this.props.vendorMap[this.state.vendor]
+                .map(os => {
+                    return (<Select.SelectEntry data={os.shortId} key={os.shortId}>{os.name}</Select.SelectEntry>);
+                })
+        )
+
+        let units = ([
+            <Select.SelectEntry data='MiB' key='MiB'>MiB</Select.SelectEntry>,
+            <Select.SelectEntry data='GiB' key='GiB'>GiB</Select.SelectEntry>
+        ])
+
+
+        return (
+            <div className="modal-body modal-dialog-body-table">
+                <table className="form-table-ct">
+                    <tr>
+                        <td className="top">
+                            <label className="control-label" for="vm-name">
+                                {_("Name")}
+                            </label>
+                        </td>
+                        <td>
+                            <input id="vm-name" className="form-control" type="text" minLength={1}
+                                   value={this.props.vmParams.vmName}
+                                   onChange={this.onChangedEventValue.bind(this, 'vmName')}/>
+                        </td>
+                    </tr>
+                    <tr>
+                        <td className="top">
+                            <label className="control-label" for="location">
+                                {_("Installation Location")}
```

Comment:
At first I thought this referred to what directory it would install the OS into.
I would make this "Installation Source" to be more consistent with the Fedora/RHEL installer.
See https://access.redhat.com/webassets/avalon/d/Red_Hat_Enterprise_Linux-7-Installation_Guide-en-US/images/2f2c6dc8424ea84cc783f1d26d645839/installationsource.png